cogeo_mosaic/backends/base.py

```diff
@@ -18,18 +19,57 @@
 
 from cogeo_mosaic.backends.utils import find_quadkeys, get_hash
 from cogeo_mosaic.cache import cache_config
-from cogeo_mosaic.errors import NoAssetFoundError
+from cogeo_mosaic.errors import NoAssetFoundError, UnsupportedOperation
 from cogeo_mosaic.models import Info, Metadata
 from cogeo_mosaic.mosaic import MosaicJSON
 from cogeo_mosaic.utils import bbox_union
 
 
+def update_zoom_and_bounds(instance, attribute, value):
+    """Update Zooms and Bounds attribute when mosaic_def is set."""
+    instance.minzoom = value.minzoom
+    instance.maxzoom = value.maxzoom
+    instance.bounds = value.bounds
+    return value
```

Comment:
This is a `magic` method which will update the instance zoom/bounds each time `mosaic_def` is assigned to a new value

Reply:
I think it might be a little clearer if this were a static method on the 
`BaseBackend` class (and maybe `mode_validator` too). That would also allow these to be subclassed if necessary for some reason.

Reply:
let me retry but I think last time I tried those method weren't passed to subclasses, so I had to re-write the same code. having those outside the class just help for reusability